tests/test_create_pod_attaching_multiple_pvc_start_IO.py

```diff
@@ -0,0 +1,376 @@
+import os
+import yaml
+import logging
+import base64
+import pytest
+import ocs.ocp
+import ocs.pod
+import ocs.defaults as defaults
+
+from ocs import exceptions
+from time import sleep
+from ocsci import tier1, ManageTest
+from ocsci.enums import StatusOfTest
+from utility import templating
+
+log = logging.getLogger(__name__)
+
+CBP = ocs.ocp.OCP(
+    kind='CephBlockPool', namespace=defaults.ROOK_CLUSTER_NAMESPACE
+)
+SEC = ocs.ocp.OCP(
+    kind='Secret', namespace=defaults.ROOK_CLUSTER_NAMESPACE
+)
+SC = ocs.ocp.OCP(
+    kind='StorageClass', namespace=defaults.ROOK_CLUSTER_NAMESPACE
+)
+PVC = ocs.ocp.OCP(
+    kind='PersistentVolumeClaim', namespace=defaults.ROOK_CLUSTER_NAMESPACE
+)
+POD = ocs.ocp.OCP(
+    kind='Pod', namespace=defaults.ROOK_CLUSTER_NAMESPACE
+)
+OCP = ocs.ocp.OCP(
+    kind='Service', namespace=defaults.ROOK_CLUSTER_NAMESPACE
+)
+DC = ocs.ocp.OCP(
+    kind='DeploymentConfig', namespace=defaults.ROOK_CLUSTER_NAMESPACE
+)
+
+TEMP_YAML = os.path.join("templates/ocs-deployment", "temp.yaml")
+TEMPLATES_DIR = "templates/CSI/rbd"
+CIRROS_DIR = "templates/app-pod-yamls"
+
+
+@pytest.fixture(scope='class')
+def test_fixture(request):
+    """
+    Removing the resources created for this test
+    """
+    self = request.node.cls
+
+    def finalizer():
+        teardown(self)
+    request.addfinalizer(finalizer)
+    setup(self)
+
+
+def setup(self):
+    """
+    Setting up the environment for the test
+    """
+    create_ceph_block_pool()
+    create_rbd_secret()
+    create_rbd_storage_class()
+
+
+def teardown(self):
+    """
+    Teardown the created Environment
+    """
+    log.info(f"Cirros pod name {self.cirros_name[0]}")
+    delete_pod(self.cirros_name[0])
+    delete_persistent_volume(self.pv_list)
+    delete_storage_class()
+    delete_pool()
+    delete_secret()
+    os.remove(TEMP_YAML)
+
+
+def create_ceph_block_pool():
+    """
+    Create ceph block pool
+
+    """
+    template = os.path.join(TEMPLATES_DIR, "CephBlockPool.yaml")
+    log.info(f"Create a ceph block pool")
+    data = templating.generate_yaml_from_jinja2_template_with_data(
+        template
+    )
+    ceph_block_pool = data['metadata']['name']
+    with open(TEMP_YAML, 'w') as yaml_file:
+        yaml.dump(data, yaml_file)
+    assert CBP.create(yaml_file=TEMP_YAML)
+    open(TEMP_YAML, 'w').close()
+    sleep(5)
```

Comment:
What is actually created as part of the cephblockpool?
I don't like the usage of sleep

Reply:
Immediately after "oc create -f <yaml>" observing some level of inconsistency. for eg: couple of runs verification of service creation passes and sometimes it fails, in case of failure the service will be created in OCP looks like some timing problem. 
for pods wait_for_resource function works, to have a timeout but get() function doens't have wait or to re-try for the service.